api/core/v1beta2/cluster_types.go

```diff
@@ -1184,8 +1185,8 @@ type Cluster struct {
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	// spec is the desired state of Cluster.
-	// +optional
-	Spec ClusterSpec `json:"spec,omitempty"`
+	// +required
```

Comment:
required because either Cluster.spec.topology or Cluster.spec.controlPlaneRef / infrastructureRef should be set. An empty Cluster does not work

Reply:
No, there is no validation today. Neither in OpenAPI validation nor in the webhook.

Reply:
We could probably build some ratcheting validation to enforce this in the future, something for the backlog maybe

Reply:
Yeah, will track as a follow-up to improve this